pymc/tuning/starting.py

```diff
@@ -120,7 +120,8 @@ def find_MAP(
     compiled_logp_func = DictToArrayBijection.mapf(model.compile_logp(jacobian=False), start)
     logp_func = lambda x: compiled_logp_func(RaveledVars(x, x0.point_map_info))
 
-    rvs = [model.values_to_rvs[value] for value in vars]
+    vars_dict = {var.name: var for var in vars}
```

Comment:
we could replace the `var_names` set above with this dict so that we don't create two objects with similar purposes.